src/operator/tensor/elemwise_unary_op_basic.cc

```diff
@@ -399,6 +399,37 @@ NNVM_REGISTER_OP(reshape_like)
 .add_argument("lhs", "NDArray-or-Symbol", "First input.")
 .add_argument("rhs", "NDArray-or-Symbol", "Second input.");
 
+NNVM_REGISTER_OP(shape)
+.describe("Returns a 1D int64 array containing the shape of data.")
+.set_num_inputs(1)
+.set_num_outputs(1)
+.set_attr<nnvm::FInplaceIdentity>("FInplaceIdentity",
+    [](const NodeAttrs& attrs){ return std::vector<bool>{true}; })
+.set_attr<nnvm::FIgnoreInputs>("FIgnoreInputs",
+    [](const NodeAttrs& attrs) { return std::vector<uint32_t>(1, 1); })
+.set_attr<FCompute>("FCompute<cpu>", ShapeCompute<cpu>)
+.set_attr<nnvm::FInferShape>("FInferShape",
+	[](const nnvm::NodeAttrs& attrs,
+	   std::vector<TShape> *in_attrs,
+	   std::vector<TShape> *out_attrs) {
+	     CHECK_EQ(in_attrs->size(), 1U);
+	     CHECK_EQ(out_attrs->size(), 1U);
+	     TShape target_shape(1);
+	     target_shape[0] = in_attrs->at(0).ndim();
+	     SHAPE_ASSIGN_CHECK(*out_attrs, 0, target_shape);
+	     return out_attrs->at(0).ndim() != 0U && out_attrs->at(0).Size() != 0U;
```

Comment:
use `shape_is_none(out_attrs->at(0)`

Reply:
you suggest that I define a shape_is_none function and use it here?

Reply:
It's already available.